src/Nethermind/Nethermind.Init.Snapshot/InitDatabaseSnapshot.cs

```diff
@@ -71,44 +71,53 @@ private async Task InitDbFromSnapshot(CancellationToken cancellationToken)
 
         Directory.CreateDirectory(snapshotConfig.SnapshotDirectory);
 
-        if (GetCheckpoint(snapshotConfig) < Stage.Downloaded)
+        if (!File.Exists(snapshotUrl))
```

Comment:
Url and file? Did you mean `snapshotFileName`? Would be good to abstract file manipulation to some methods instead of dign it directly.